modules/EmergencyMedicine/config/module.config.php

```diff
@@ -25,23 +27,72 @@
 
     'controllers' => array(
         'factories' => [
+            letterGeneratorController::class => function (ContainerInterface $container, $requestedName) {
+                return new letterGeneratorController($container);
+            },
 
-
+            xrayLetterController::class => function (ContainerInterface $container, $requestedName) {
+                return new xrayLetterController($container);
+            },
         ],
-
     ),
 
     /**
      * routing configuration.
-     * for more option and details - http://zf2.readthedocs.io/en/latest/in-depth-guide/understanding-routing.html?highlight=routing
      */
     'router' => array(
         'routes' => array(
+
+            /****************************************************************/
+            'letter_generator' => array(
+                'type'    => 'segment',
+                'options' => array(
+                    'route'    => '/letter-generator[/:action][/:id]',
+                    'constraints' => array(
+                        'action' => '[a-zA-Z][a-zA-Z0-9_-]*',
+                        'method'     => '[a-zA-Z][a-zA-Z0-9_-]*',
+                    ),
+                    'defaults' => array(
+                        'controller' => letterGeneratorController::class,
+                        'action'     => 'index',
+                    ),
+                ),
+            ),
+            /****************************************************************/
+            'xray_letter' => array(
```

Comment:
Is it needed ????
We are using API connection...

Reply:
the type of connection is not important , without this mapping the API does not recognize the class

Reply:
yes it is needed other wise the api can not call the controller regardless of the type of connection